src/main/java/com/microsoft/sqlserver/jdbc/SimpleTtlCache.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.time.Duration;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Internal class implements a cache data structure based on ConcurrentHashMap. The cache uses a scheduler to 
+ * provide eviction for the cache entry according to the pre-defined time-to-live value.
+ * 
+ * This class is used as local caches for the column encrypted keys and the results of signature verification of 
+ * column master key metadata. Refer to SQLServerColumnEncryptionAzureKeyVaultProvider for an example.
+ */
+final class SimpleTtlCache<K,V> {    
+
+    // This class clears cache entry when it is called by scheduler. 
+    class CacheClear implements Runnable {
+    
+        private K keylookupValue;
+        final private java.util.logging.Logger logger = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.SimpleTtlCache.CacheClear");
+    
+        CacheClear(K keylookupValue) {
+            this.keylookupValue = keylookupValue;
+        }
+    
+        @Override
+        public void run() {
+            // remove() is a no-op if the key is not in the map.
+            // It is a concurrentHashMap, update/remove operations are thread safe.
+            if (cache.containsKey(keylookupValue)) {
+                V value = cache.get(keylookupValue);
+                
+                if (value instanceof SQLServerSymmetricKey) {
+                    ((SQLServerSymmetricKey) value).zeroOutKey();
+                }
+                
+                cache.remove(keylookupValue);
+                if (logger.isLoggable(java.util.logging.Level.FINE)) {
+                    logger.fine("Removed key from cache...");
+                }
+            }
+        }
+    }
+
+    static final private java.util.logging.Logger simpleCacheLogger = java.util.logging.Logger.getLogger("com.microsoft.sqlserver.jdbc.SimpleTtlCache");
+
+    private final ConcurrentHashMap<K, V> cache;
+    private Duration cacheTtl = Duration.ofHours(2);  // The default time-to-live is set to 2 hours.

```

Comment:
can we define a constant for this at the top, makes it easier to see the hard code value

Reply:
removed